<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.40"/>
  <arg name="y_pos" default="0.4"/>
  <arg name="z_pos" default="0.05"/>
  <arg name="maze" default="maze2"/>

  <include file="$(find RBE3002_template)/launch/maze_sim.launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="x_pos" value="$(arg x_pos)"/>
    <arg name="y_pos" value="$(arg y_pos)"/>
    <arg name="z_pos" value="$(arg z_pos)"/>
    <arg name="maze" value="$(arg maze)"/>
  </include>

  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <!--gmapping. We don't need mapserver if gmapping is running -->
  <include file="$(find turtlebot3_slam)/launch/turtlebot3_gmapping.launch" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rbe3002_lab4)/rviz/rbe3002_lab4.rviz"/>

  <node name="drive_server" pkg="rbe3002_lab4" type="drive_server.py" args="-d $(find turtlebot3_gazebo)/turtlebot3_gazebo/launch/turtlebot3_lab4_simulation.launch"/>

  <node name="frontier_finder" pkg="rbe3002_lab4" type="frontier_finder.py" args="-d $(find turtlebot3_gazebo)/turtlebot3_gazebo/launch/turtlebot3_lab4_simulation.launch"/>

  <node name="path_planner" pkg="rbe3002_lab4" type="path_planner.py" args="-d $(find turtlebot3_gazebo)/turtlebot3_gazebo/launch/turtlebot3_lab4_simulation.launch"/>

  <node name="main" pkg="rbe3002_lab4" type="main.py" args="-d $(find turtlebot3_gazebo)/turtlebot3_gazebo/launch/turtlebot3_lab4_simulation.launch"/>

</launch>
